cmd/zfs_object_agent/src/zettacache.rs

```diff
@@ -1076,19 +1077,19 @@ impl ZettaCacheState {
         // Note: the caller is about to flush as well, but we want to count the time in the below info!
         new_index.flush().await;
 
-        debug!("new histogram starts at {}: {:#?}", eviction_atime, new_index.atime_histogram);
-        // XXX - this will no longer be true if we are evicting some blocks.
-        // The index histogram will be the "tail"
-        //assert_eq!(
-        //    new_index.atime_histogram.histogram,
-        //    self.atime_histogram.histogram
-        //);
+        // Free the evicted blocks from the cache
+        debug!("freeing {} blocks", new_index.free_list.len());
+        while let Some(value) = new_index.free_list.pop() {
```

Comment:
I think it's more straightforward to use the iterator (which can even be the consuming iterator in this case; if that doesn't work then you can use the reference-iterator with `in free_list.iter()` or `in &free_list`):

```suggestion
        for value in new_index.free_list {
```

And then after that, `new_index.free_list = Vec::new();`, if you take my suggestion above of having free_list be a local variable, you don't need to do that (you just don't access free_list after consuming it with the iterator).

Reply:
Done